test/unit/css.js

```diff
@@ -664,6 +654,190 @@ QUnit.test( "show() after hide() should always set display to initial value (#14
 	assert.equal( div.css( "display" ), "list-item", "should get last set display value" );
 } );
 
+QUnit.test( "show/hide 3.0, default display", function( assert ) {
+
+	assert.expect( 36 );
+
+	var i,
+		$elems = jQuery( "<div/>" )
+			.appendTo( "#qunit-fixture" )
+			.html( "<div title='block'/><span title='inline'/><ul><li title='list-item'/></ul>" )
+			.find( "[title]" );
+
+	$elems.each( function() {
+		var $elem = jQuery( this ),
+			name = this.nodeName + this.className.replace( /(?=.)/, "." ) +
+				( this.getAttribute( "style" ) || "" ).replace( /.+/, "[style='$&']" ),
```

Comment:
Yeah, please make it more simpler, pretty convoluted. I'd say tests is not the place where we need to be clever